File: cpp/WordFilterTest.cpp
/*
 * =====================================================================================
 *
 *       Filename: WordFilterTest.cpp
 *
 *    Description:  
 *
 *        Version:  1.0
 *        Created:  11/17/2012 12:03:47 AM
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  witwolf (ALL), wo dot bian dot shi dot mo0 at gmail dot com
 *        Company:  ...
 *
 * =====================================================================================
 */

#include "WordFilter.h"
#include <iostream>
using namespace std;
int main()
{
    char* _words[] = 
	{
	    "fuck you"
	    ,"aad"
	    ,"ac"
	    ,"ababd"
	    ,"abc"
	    ,"you"
	};
    string src = "aac fuck You ababc haha aaababd" ;
    vector<string> words;
    for(int i = 0 ; i < sizeof(_words)/sizeof(char*) ;i++)
	{
	    words.push_back(_words[i]);
	}
   
    WordFilter wordFilter;
    wordFilter.initFilterWithWords(words);

    string ret = wordFilter.applyFilter(src);

    cout << "before filter : " << src << endl ;
    cout << "after filter : " << ret << endl ;
}

File: java/WordFilterTest.java
import java.util.List;
import java.util.LinkedList;
public class WordFilterTest {
	public static void main(String args[])
	{
		String [] _words = 
			{
				"ababd",
				"fuck",
				"abc",
				"fuck you"
			};
		
		List<String> words = new LinkedList<String>();
		for(String word:_words)
		{
			words.add(word);
		}
		
		
		WordFilter wordFilter = new WordFilter();
		wordFilter.initFilterWithWords(words);
		
		String src = "ababc Fuck yo";
		String ret = wordFilter.AfterFilter(src);
		
		System.out.println("before filter : " + src);
		System.out.println("after filter : " + ret);
		
	}

}
